labs/a4.md

```diff
@@ -0,0 +1,164 @@
+order to get checked off for this portion of the lab, ensure the following:
+
+1. All incoming connections are blocked by default.
+2. Allow incoming connections to your VM over port 22 (SSH), otherwise you'll be
+   locked out of your VM (as will we).
+3. Allow incoming connections for the ports necessary for the operation of the
+   software you'll be installing later on in this lab, e.g. port 80 and/or 443
+   for a web server.
+4. Bonus points: Allow the above, but only for IPs originating from UC Berkeley's
+   subnet. UC Berkeley has 3 primary `/16`s.
+5. Double bonus points: Configure `fail2ban` to block IP addresses that are trying
+   to brute-force your SSH password.
+
+_Checkoff: paste the output of `sudo ufw status verbose`_
+
+## Choose your own adventure
+
+At this stage, your task is to install a useful service on your VM. Here are some
+suggestions, but feel free to set up anything you'd like. Your submission will
+consist of explaining what you installed, and explaining any potential security
+implications you can think of. You don't need to do all of them.
+
+### Music Server
+
+Some music enthusiasts opt to own their own music files instead of using a cloud
+service (Spotify, Google Play Music, etc). But, if the files are only stored on
+your computer’s hard drive, how can you listen to your music when on someone else’s
+computer or on the go? A music server like `CherryMusic` can solve this problem
+while still giving you control over your files. Install
+[CherryMusic](http://www.fomori.org/cherrymusic/) (or an
+[alternative](https://github.com/Kickball/awesome-selfhosted#audio-streaming)]) on your
+DeCal VM, and load up a few tunes.
+
+### Git server
+
+GitHub is a good place to host your source code, but self-hosting your own Git
+can be a good option if you need to store something private (and don’t trust GitHub),
+hosting your own can be a good solution. While you can use Git to host a repo on any
+SSH server using a [bare repo](http://www.saintsjd.com/2011/01/what-is-a-bare-git-repository/),
+to get a fancy GUI you need to install special software. Some popular options
+for this include [Gitea](https://gitea.io/en-us/) (which mimicks GitHub’s interface)
+and [cgit](https://git.zx2c4.com/cgit/about/) (which takes a more minimalist approach).
+Pick one and install it.
+
+### File Hosting
+
+Instead of uploading your files to Google Drive or Dropbox to be shared, you can
+host on your own VM. A good option for this is [fluffy](https://github.com/chriskuehl/fluffy)
+(created by an OCF alum!), which also doubles as a pastebin for quickly sharing code snippets.
+For more options, look at the the [“simple click/drag-n-drop upload” section of
+[awesome-selfhosted](https://github.com/Kickball/awesome-selfhosted#single-clickdrag-n-drop-upload).
+
+### Password Manager
+
+[BitWarden](https://blog.bitwarden.com/host-your-own-open-source-password-manager-ace147649936) is
```

Comment:
noice. perhaps add a short note that you should _not_ store your password on your student VM, as you will lose access at the end of the semester, and decal staff have access (although not to the password directly)